### PR TITLE
feat: use `atlas` in `make pull_translations` on palm FC-0012

### PR DESCRIPTION
**Pull request by OmarIthawi**
This is a cherry-pick of #124 into `open-release/palm.master` to fix translations for the Communications MFE and enable translations on the Tutor MFE for this app: 

 - https://github.com/overhangio/tutor-mfe/pull/149

Changes
-------
 - Bump frontend-platform to bring intl-imports.js script
 - Move all i18n imports into `src/i18n/index.js` so intl-imports.js can override it with latest translations
 - Add `atlas` into `make pull_translations` when `OPENEDX_ATLAS_PULL` environment variable is set.
 - Fixed lint rules for frontend-platform@4.1.0
 - Mock useTrackColorSchemeChoice to avoid test failures
 - Add openedx-atlas
 - Deprecate all other Transifex broken functionality

## TODO

 - [x] Test with the `tutor-mfe` plugin

Reference
-----

This contribution is part of the [FC-0012 project](https://openedx.atlassian.net/l/cp/XGS0iCcQ) which is sparked by the [Translation Infrastructure update OEP-58](https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0058-arch-translations-management.html#specification).

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @OmarIthawi!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Review by brian-smith-tcril — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -42,9 +43,24 @@ push_translations:
 	# Pushing comments to Transifex...
 	./node_modules/@edx/reactifex/bash_scripts/put_comments_v3.sh
 
+ifeq ($(OPENEDX_ATLAS_PULL),)
```
Comment (OmarIthawi) [inline Makefile:46]: Remove the `if`, it makes no sense here.

File: Makefile [hunk 2]
```diff
@@ -31,20 +29,17 @@ detect_changed_source_translations:
 	# Checking for changed translations...
 	git diff --exit-code $(i18n)
 
-# Pushes translations to Transifex.  You must run make extract_translations first.
-push_translations:
-	# Pushing strings to Transifex...
-	tx push -s
```
Comment (OmarIthawi) [inline Makefile:37]: It makes no sense to keep a broken step that has no Transifex input.

**Comment by OmarIthawi** [comment]
@brian-smith-tcril @arbrandes would you mind merging this pull request if you think it's ready?

**Comment by OmarIthawi** [comment]
@arbrandes @brian-smith-tcril should we merge this branch into Quince? At the moment it's targeting the Palm branch.

**Comment by brian-smith-tcril** [comment]
@OmarIthawi I think it probably makes sense to keep this PR *and* to create a new one targeting Quince.

**Comment by OmarIthawi** [comment]
@arbrandes would you mind merging this pull request if you think it's ready?

**Comment by openedx-webhooks** [comment]
@OmarIthawi 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-app-communications&entry.867055334=https://github.com/openedx/frontend-app-communications/pull/156&entry.1484655318=https://github.com/OmarIthawi&entry.752974735=2023-10-10+11:48&entry.1917517419=2023-11-03+20:57&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**